client/src/main/java/org/ehrbase/client/openehrclient/builder/ContributionBuilder.java

```diff
@@ -0,0 +1,401 @@
+import static org.ehrbase.client.openehrclient.ContributionChangeType.*;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.nedap.archie.json.JacksonUtil;
+import com.nedap.archie.rm.RMObject;
+import com.nedap.archie.rm.changecontrol.OriginalVersion;
+import com.nedap.archie.rm.generic.AuditDetails;
+import com.nedap.archie.rm.support.identification.ObjectVersionId;
+import java.util.LinkedHashMap;
+import java.util.Map;
+import javax.annotation.Nullable;
+import org.ehrbase.client.openehrclient.ContributionChangeType;
+import org.ehrbase.response.openehr.ContributionCreateDto;
+import org.ehrbase.serialisation.dbencoding.RmObjectEncoding;
+import org.ehrbase.serialisation.jsonencoding.CanonicalJson;
+
+public class ContributionBuilder {
+    private final ContributionCreateDto contributionCreateDto;
+
+    public static ContributionBuilderDtoBuilder builder(AuditDetails audit) {
+        return new ContributionBuilderDtoBuilder(audit);
+    }
+
+    public ContributionBuilder(ContributionCreateDto contributionCreateDto) {
+        this.contributionCreateDto = contributionCreateDto;
+    }
+
+    /**
+     * Gets contribution dto.
+     *
+     * @return the contribution create dto
+     */
+    public ContributionCreateDto getContribution() {
+        return contributionCreateDto;
+    }
+
+    public static class ContributionBuilderDtoBuilder {
+
+        public static final String MANDATORY_CONTRIBUTOR_AUDIT_DETAILS = "Missing mandatory contributor AuditDetails.";
+
+        public static final String MISSING_MANDATORY_PRECEDING_VERSION_UID = "Missing mandatory precedingVersionUid.";
+        public static final String INVALID_PRECEDING_VERSION_UID_IN_VERSION_CONTAINER =
+                "Input invalid. Composition can't be modified without pointer to precedingVersionUid in Version container.";
+
+        private final ContributionCreateDto contributionCreateDto = new ContributionCreateDto();
+
+        /**
+         * Instantiates a new Contribution.
+         *
+         * @param audit The audit for contribution and compositions inside
+         */
+        ContributionBuilderDtoBuilder(AuditDetails audit) {
+            this.contributionCreateDto.setAudit(audit);
+        }
+
+        /**
+         * Add composition creation change type to contribution.
+         *
+         * @param composition RMObject composition instance
+         * @return the contribution create dto
+         */
+        public ContributionBuilderDtoBuilder addCompositionCreation(final RMObject composition) {
+            LinkedHashMap<String, Object> canonicalComposition = getCanonicalComposition(composition);
+            updateContribution(canonicalComposition, CREATION, null);
+
+            return this;
+        }
+
+        /**
+         * Add composition creation change type to contribution.
+         *
+         * @param composition the composition
+         * @return the contribution create dto
+         */
```

Comment:
I think you can change that to set the RM Object here and it should work with out converting it to a map.

Reply:
Also since you just tested compositions, and not folder or erhstatus. Which is ok for the first implementation. then you should here just allow compositions

Reply:
done